### PR TITLE
Accept partial copilot suggestions

### PR DESCRIPTION
**Pull request by jdk-21**
Release Notes:

- Added ability to accept partial copilot suggestions (#8020)
 
This PR adds a new shortcut cmd-right, if a copilot suggestion exists. The suggestions is accepted word by word.
It emulates the behaviour of VS Code's Github Copilot implementation.

This is my first PR in Rust so I'm glad to receive feedback.

### CODE REVIEW

**Comment by jdk-21** [comment]
@cla-bot check

File: assets/keymaps/default-macos.json [hunk 1]
```diff
@@ -190,6 +190,12 @@
       "ctrl-shift-enter": "editor::NewlineBelow"
     }
   },
+  {
+    "context": "Editor && copilot_suggestion",
```
Comment (SomeoneToIgnore) [inline assets/keymaps/default-macos.json:194]: https://github.com/zed-industries/zed/blob/e6bec9a24a12773fc8e9d4ed0a4d37cb06bed75c/assets/keymaps/default-linux.json#L138-L140

We seem to have a group of copilot keybindings, starting with `alt` and `alt-right` is vacant, so let's regroup the whole thing?

Seems that existing copilot keybindings should work in any `Editor` context, and only `copilot::Suggest` should have `!copilot_suggestion`, we can create another block for the `copilot::Suggest` and move the rest of the existing keybindings here.

Let's also change the `cmd` to `alt` to match the rest of the bindings and do the same changes to `default-linux.json`

File: crates/editor/src/editor.rs [hunk 2]
```diff
@@ -3969,6 +3973,39 @@ impl Editor {
         }
     }
 
+    fn accept_partial_copilot_suggestion(
+        &mut self,
+        _: &AcceptPartialCopilotSuggestion,
```
Comment (SomeoneToIgnore) [inline crates/editor/src/editor.rs:3978]: We have a set of tests on copilot already: https://github.com/zed-industries/zed/blob/e6bec9a24a12773fc8e9d4ed0a4d37cb06bed75c/crates/editor/src/editor_tests.rs#L7392

That test function is quite large, but we should be able to create a new one with the same LSP set-up, testing the new code, let's do that.

**Review by SomeoneToIgnore — COMMENTED** [review]
Thank you, that works really nice!
Seems that it's almost ready to merge, only a couple of places to fix first.

File: assets/keymaps/default-macos.json (hunk 1 above)
Comment (jdk-21) [reply assets/keymaps/default-macos.json:194]: Seems like a good idea. But right now alt-right is used for moving to the end of a word. Maybe we should do the blocks like this:
```
{
  "context": "Editor && copilot_suggestion",
  "bindings":
      "alt-]": "copilot::NextSuggestion",
      "alt-[": "copilot::PreviousSuggestion",
      "alt-right": "editor::AcceptPartialCopilotSuggestion",
    }
  },
  {
    "context": "Editor && !copilot_suggestion",
    "bindings": {
      "alt-\\": "copilot::Suggest"
    }
  }
```

Comment (SomeoneToIgnore) [reply assets/keymaps/default-macos.json]: Right, how did I manage to miss the `MoveToNextWordEnd`? 😕 
Still, looks like it works just fine in the context I've proposed, and we do not propose copilot suggestions that are not on the "edge" of a block, so seems ok to keep it this way maybe.

We can change it later if somebody comes with a good example that breaks the workflow.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you!

**Pull request merged**